src/hotspot/share/classfile/modules.cpp

```diff
@@ -649,17 +652,20 @@ void Modules::serialize_native_access_flags(SerializeClosure* soc) {
 }
 
 void Modules::dump_addmods_names() {
+  ResourceMark rm;
   const char* addmods_names = get_addmods_names_as_sorted_string();
   if (addmods_names != nullptr) {
     _archived_addmods_names = ArchiveBuilder::current()->ro_strdup(addmods_names);
   }
 }
 
 const char* Modules::get_addmods_names_as_sorted_string() {
+  assert(Thread::current()->current_resource_mark() != nullptr, "Setup by caller");
   return get_numbered_property_as_sorted_string("jdk.module.addmods");
 }
 
 void Modules::serialize_addmods_names(SerializeClosure* soc) {
+  ResourceMark rm;
```

Comment:
Again I would suggest moving this to the scope where it is needed.

Reply:
I think it is right scope.